### PR TITLE
CLOUD-374 PMM2.0 integration

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

File: pkg/psmdb/pmm.go [hunk 1]
```diff
@@ -12,7 +12,12 @@ const (
 )
 
 // PMMContainer returns a pmm container from given spec
-func PMMContainer(spec api.PMMSpec, secrets string, customLogin bool) corev1.Container {
+func PMMContainer(spec api.PMMSpec, secrets string, customLogin bool, clusterName string) corev1.Container {
+	ports := []corev1.ContainerPort{{ContainerPort: 7777}}
+
+	for i := 30100; i <= 30200; i++ {
```
Comment (delgod) [inline pkg/psmdb/pmm.go]:
```suggestion
	for i := 30100; i <= 30005; i++ {
```

File: pkg/psmdb/pmm.go [hunk 2]
```diff
@@ -50,9 +55,22 @@ func PMMContainer(spec api.PMMSpec, secrets string, customLogin bool) corev1.Con
 			},
 			{
 				Name:  "DB_ARGS",
-				Value: "--uri=mongodb://$(MONGODB_USER):$(MONGODB_PASSWORD)@127.0.0.1:27017/",
+				Value: "--uri=mongodb://$(MONGODB_USER):$(MONGODB_PASSWORD)@127.0.0.1:27017/ --use-profiler",
```
Comment (delgod) [inline pkg/psmdb/pmm.go]: is this option needed to PMM2 ?

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 3]
```diff
@@ -117,9 +117,11 @@ type ClusterCondition struct {
...
 
 type PMMSpec struct {
-	Enabled    bool   `json:"enabled,omitempty"`
-	ServerHost string `json:"serverHost,omitempty"`
-	Image      string `json:"image,omitempty"`
+	Enabled        bool   `json:"enabled,omitempty"`
+	ServerHost     string `json:"serverHost,omitempty"`
+	Image          string `json:"image,omitempty"`
+	ServerUser     string `json:"serverUser,omitempty"`
+	ServerPassword string `json:"serverPassword,omitempty"`
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_types.go]: why this change needed?

File: pkg/controller/perconaservermongodb/secrets.go [hunk 4]
```diff
@@ -48,6 +48,11 @@ func (r *ReconcilePerconaServerMongoDB) reconcileUsersSecret(cr *api.PerconaServ
 	if err != nil {
 		return fmt.Errorf("create admin users pass: %v", err)
 	}
+	if len(cr.Spec.PMM.ServerUser) > 0 && len(cr.Spec.PMM.ServerPassword) > 0 {
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/secrets.go]: why this change needed?

File: pkg/psmdb/pmm.go [hunk 5]
```diff
@@ -50,9 +55,22 @@ func PMMContainer(spec api.PMMSpec, secrets string, customLogin bool) corev1.Con
...
+				Name:  "DB_HOST",
+				Value: "localhost",
+			},
+			{
+				Name:  "DB_CLUSTER",
+				Value: clusterName,
+			},
+			{
+				Name:  "DB_PORT",
+				Value: "27017",
```
Comment (delgod) [inline pkg/psmdb/pmm.go]: please also pass min and max ports

File: pkg/psmdb/pmm.go (hunk 1 above)
Comment (cap1984) [reply pkg/psmdb/pmm.go]: done by @nonemax

File: pkg/psmdb/pmm.go (hunk 2 above)
Comment (cap1984) [reply pkg/psmdb/pmm.go]: nothing more since RC3
done by @nonemax

File: pkg/apis/psmdb/v1/psmdb_types.go (hunk 3 above)
Comment (cap1984) [reply pkg/apis/psmdb/v1/psmdb_types.go]: outdated

File: pkg/controller/perconaservermongodb/secrets.go (hunk 4 above)
Comment (cap1984) [reply pkg/controller/perconaservermongodb/secrets.go]: outdated

File: pkg/psmdb/pmm.go (hunk 5 above)
Comment (cap1984) [reply pkg/psmdb/pmm.go]: done by @nonemax

File: pkg/psmdb/pmm.go [hunk 6]
```diff
@@ -12,7 +12,85 @@ const (
...
+					Key: "MONGODB_CLUSTER_MONITOR_PASSWORD",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secrets,
+					},
+				},
+			},
+		},
+		{
+			Name:  "DB_ARGS",
+			Value: "--uri=mongodb://$(MONGODB_USER):$(MONGODB_PASSWORD)@127.0.0.1:27017/ --use-profiler",
```
Comment (delgod) [inline pkg/psmdb/pmm.go]:
```suggestion
			Value: "--uri=mongodb://$(MONGODB_USER):$(MONGODB_PASSWORD)@127.0.0.1:27017/",
```

Comment (cap1984) [reply pkg/psmdb/pmm.go]: done

File: e2e-tests/monitoring/conf/monitoring-rs0.yml [hunk 7]
```diff
@@ -23,7 +23,7 @@ spec:
     size: 3
   pmm:
     enabled: true
-    image: perconalab/pmm-client:1.17.1
+    image: -pmm
```
Comment (cap1984) [inline e2e-tests/monitoring/conf/monitoring-rs0.yml]: typo?

File: e2e-tests/one-pod/conf/one-pod-rs0.yml [hunk 8]
```diff
@@ -11,7 +11,7 @@ spec:
     users: some-users
   pmm:
     enabled: false
-    image: perconalab/pmm-client:1.17.1
+    image: -pmm
```
Comment (cap1984) [inline e2e-tests/one-pod/conf/one-pod-rs0.yml]: typo?

File: e2e-tests/monitoring/conf/monitoring-rs0.yml (hunk 7 above)
Comment (delgod) [reply e2e-tests/monitoring/conf/monitoring-rs0.yml]: no, please look regex in 514 line in e2e-tests/functions

Comment (cap1984) [reply e2e-tests/monitoring/conf/monitoring-rs0.yml]: ок
